src/common/controls/Control.js

```diff
@@ -1,11 +1,54 @@
-/* eslint-disable class-methods-use-this */
+import Observable from 'ol/Observable';
 
-class Control {
-  activate(map) {
+/**
+ * Generic control for mobility-toolbox-js.
+ */
+class Control extends Observable {
+  /**
+   * Constructor
+   *
+   * @param {ol/Map~Map|mapboxgl.Map} map Control's map.
+   * @param {Object} [options] Control options.
+   * @param {boolean} [options.active = true] Whether the control is active.
+   */
+  constructor(map, options = {}) {
+    super();
     this.map = map;
+
+    /** @ignore */
+    this.options = {
+      active: true,
+      ...options,
+    };
+
+    this.active = options.active;
+
+    if (this.options.active) {
+      this.activate();
+    }
+  }
+
+  /**
+   * Activate the control.
+   */
+  activate() {
+    this.active = true;
+  }
+
+  /**
+   * Deactivate the control.
+   */
+  deactivate() {
+    this.active = false;
   }
 
-  deactivate() {}
+  /**
+   * Get te control's active state.
+   * @returns {boolean} Active state.
+   */
+  getActive() {
```

Comment:
no need of this . Access directly the active property is enough.  You could use definePropertes method like in the Layers to define property, see https://github.com/geops/mobility-toolbox-js/blob/master/src/common/layers/Layer.js#L87

And if we extend ol.Control we could just define the property like this so an event will be triggered automatically on change.

So we could remove activate deactivate and getActive function